src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -58,6 +61,11 @@ private CsvTupleWriter(final OutputStream out, final CSVFormat csvFormat) throws
   }
 
   @Override
+  public void open(final OutputStream stream) throws IOException {
```

Comment:
I added the open() call in cases which we wanted to open the stream from the worker side. Originally, these constructors were used in cases in which the coordinator would create the operator and open the outputStream in place (for example see getDatasetData() in DatasetResource which calls this CsvTupleWriter constructor). 

I could change this so the open() is only used in the MyriaCustomPersistenceFormatTupleWriter class. Or another alternative is to remove the public constructors (as suggested by your comment below) and keep only open() calls for all the writers.